internal/model/todos.go

```diff
@@ -19,81 +19,119 @@ type TodoList struct {
 	Todos []*Todo `json:"todos"`
 }
 
-// GetTodosListByID returns slice of Todos based on the listID passed in
-func GetTodosListByID(listID string) (*TodoList, error) {
-	listStatement := `SELECT id, name FROM lists WHERE id = $1`
+// TodoListByID fetches a list and its todos
+func TodoListByID(id string) (*TodoList, error) {
 	list := &TodoList{}
 
-	err := db.QueryRow(listStatement, listID).Scan(&list.ID, &list.Name)
+	sql := "SELECT id, name FROM lists WHERE id = $1"
+	err := db.QueryRow(sql, id).Scan(&list.ID, &list.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	sqlStatement := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 ORDER BY created_at ASC;`
+	list.Todos, err = TodosByListID(id)
 
-	rows, err := db.Query(sqlStatement, listID)
+	return list, nil
+}
+
+func (tl *TodoList) validate() error {
+	if tl.Name == "" {
+		return errors.New("name empty, must be set")
+	}
+	if len(tl.Todos) == 0 {
+		return errors.New("no todos, must have at least one")
+	}
+
+	return nil
+}
+
+// Save saves a new list to the database
+func (tl *TodoList) Save() error {
+	if err := tl.validate(); err != nil {
+		return err
+	}
+	sql := `INSERT INTO lists(name) VALUES($1) RETURNING id;`
+	err := db.QueryRow(sql, tl.Name).Scan(&tl.ID)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	for rows.Next() {
-		todo := &Todo{}
-		err := rows.Scan(&todo.ID, &todo.ListID, &todo.Todo, &todo.Done)
+	for _, t := range tl.Todos {
+		t.ListID = tl.ID
+		err := t.Save()
 		if err != nil {
```

Comment:
We defer saving the `Todo` itself here. We just have to set the `ListID` before calling it, but our validate will have told us if we forget :)

Again separating concerns here and makes the code more readable (not in this diff, but in the editor, I promise)